contracts/cctp/events/TokenMessengerEvents.sol

```diff
@@ -0,0 +1,35 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.17;
+
+abstract contract TokenMessengerEvents {
+    /**
+     * @notice Emitted when a DepositForBurn message is sent
+     * @param nonce unique nonce reserved by message
+     * @param burnToken address of token burnt on source domain
+     * @param amount deposit amount
+     * @param depositor address where deposit is transferred from
+     * @param mintRecipient address receiving minted tokens on destination domain as bytes32
+     * @param destinationDomain destination domain
+     * @param destinationTokenMessenger address of TokenMessenger on destination domain as bytes32
+     * @param destinationCaller authorized caller as bytes32 of receiveMessage() on destination domain, if not equal to bytes32(0).
+     * If equal to bytes32(0), any address can call receiveMessage().
+     */
+    event DepositForBurn(
+        uint64 indexed nonce,
+        address indexed burnToken,
+        uint256 amount,
+        address indexed depositor,
```

Comment:
do we want to use addresses or bytes32 addresses in case this ever interacts with nonEVM chains?

Reply:
These are copied from Circle CCTP contracts (and used in mocks and tests).